VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Structs/StructWithoutElementaryMethodsOverridden/StructWithoutElementaryMethodsOverriddenCodeFix.cs

```diff
@@ -0,0 +1,177 @@
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            string implementEqualsString;
+            string implementGetHashCodeString;
+            string implementToStringString;
+
+            diagnostic.Properties.TryGetValue("IsEqualsImplemented", out implementEqualsString);
+            diagnostic.Properties.TryGetValue("IsGetHashCodeImplemented", out implementGetHashCodeString);
+            diagnostic.Properties.TryGetValue("IsToStringImplemented", out implementToStringString);
+            
+            var implementEquals = bool.Parse(implementEqualsString);
+            var implementGetHashCode = bool.Parse(implementGetHashCodeString);
+            var implementToString = bool.Parse(implementToStringString);
+
+            var dict = new Dictionary<string, bool>
+                {
+                    {"Equals()", implementEquals},
+                    {"GetHashCode()", implementGetHashCode},
+                    {"ToString()", implementToString}
+                };
+
+            var statement = root.FindNode(diagnosticSpan);
+
+            context.RegisterCodeFix(CodeAction.Create(
+                string.Format(VSDiagnosticsResources.StructWithoutElementaryMethodsOverriddenCodeFixTitle, FormatMissingMembers(dict)),
+                    x => AddMissingMethodsAsync(context.Document, root, (StructDeclarationSyntax) statement,
+                            implementEquals, implementGetHashCode, implementToString),
+                    StructWithoutElementaryMethodsOverriddenAnalyzer.Rule.Id), diagnostic);
+        }
+
+        private static readonly MethodDeclarationSyntax EqualsMethod;
+        private static readonly MethodDeclarationSyntax GetHashCodeMethod;
+        private static readonly MethodDeclarationSyntax ToStringMethod;
+
+        private Task<Document> AddMissingMethodsAsync(Document document, SyntaxNode root,
+            StructDeclarationSyntax statement, bool implementEquals, bool implementGetHashCode,
+            bool implementToString)
+        {
+            var newStatement = statement;
+
+            if (!implementEquals)
+            {
+                newStatement = newStatement.AddMembers(EqualsMethod);
+            }
+
+            if (!implementGetHashCode)
+            {
+                newStatement = newStatement.AddMembers(GetHashCodeMethod);
+            }
+
+            if (!implementToString)
+            {
+                newStatement = newStatement.AddMembers(ToStringMethod);
+            }
+
+            var newRoot = root.ReplaceNode(statement, newStatement);
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+
+        private static MethodDeclarationSyntax GetEqualsMethod()
```

Comment:
These still need to be moved to static fields. I'm not sure if that message got lost earlier